Makefile.PL

```diff
@@ -23,13 +23,30 @@ requires(
     'Zonemaster::LDNS'   => 2.0,
 );
 
+# Make all platforms include inc/Module/Install/External.pm
+requires_external_bin 'find';
+if ($^O eq "freebsd") {
+    requires_external_bin 'gmake';
+};
+
 sub MY::postamble {
-        return <<'MAKE_FRAG';
-	$(MYEXTLIB): share/Makefile
-	cd share && $(MAKE) all
-MAKE_FRAG
+    my $sharemakefile = 'share/GNUmakefile';
+    my $text;
+    if ($^O eq "freebsd") {
+        # Make FreeBSD use gmake for share
+        $text = q[GMAKE ?= "gmake"] . "\n"
+            . q[pure_all :: ] . "$sharemakefile\n"
+            . "\t" . q[cd share && $(GMAKE) all] . "\n";
+    } else {
+        # Here Linux and GNU Make is assumed
+        $text = q[pure_all :: ] . "$sharemakefile\n"
+            . "\t" . q[cd share && $(MAKE) all] . "\n";
+    };
```

Comment:
Why not make both cases use the same variable for the make command?

Reply:
The $(MAKE) variable comes from the Makefile. The variable is interpolated by make.

Reply:
I think I would have gone for setting GMAKE to different values on different platforms. That way the "pure_all" clause could be the same for all platforms.

Reply:
Such as "GMAKE = $(MAKE)" and "GMAKE = gmake", respectively?

Well, it has the same effect. Please the updated code.

Reply:
Yeah, that's what I had in mind. I'm thinking that it's a little easier on the reader (i.e. us) if you don't have to compare two heavily quoted and generated makefile snippets by eye.